datacube/model/utils.py

```diff
@@ -83,10 +83,14 @@ def new_dataset_info():
     }
 
 
-def band_info(band_names):
+def band_info(band_names, uri=None, band_uris=None):
+    if band_uris is None:
+        band_uris = {name: uri if uri is not None else ''
```

Comment:
This is going to fall back to `uri` instead of `''`, which I'm not comfortable with. This will change the default behaviour for everything using `make_dataset()`, with possibly unintended consequences.